homeassistant/components/sonos/number.py

```diff
@@ -142,3 +173,357 @@ def native_value(self) -> float:
         """Return the current value."""
         to_number = LEVEL_TO_NUMBER.get(self.level_type, int)
         return cast(float, to_number(getattr(self.speaker, self.level_type)))
+
+
+class SonosGroupVolumeEntity(SonosEntity, NumberEntity):
+    """Group volume (0–100) for the player’s current group.
+
+    - Grouped: uses GroupRenderingControl to read/write group volume.
+    - Ungrouped: mirrors the player’s RenderingControl Master volume.
+    """
+
+    _attr_translation_key = "group_volume"
+    _attr_native_min_value = 0
+    _attr_native_max_value = 100
+    _attr_native_step = 1
+    _attr_mode = NumberMode.SLIDER
+
+    def __init__(self, speaker: SonosSpeaker, config_entry: SonosConfigEntry) -> None:
+        """Initialize the Sonos group volume number entity."""
+        super().__init__(speaker, config_entry)
+        self._attr_unique_id = f"{self.soco.uid}-group_volume"
+
+        self._coord_uid: str | None = None
+        self._group_uid: str | None = None
+
+        self._unsubscribe_coord: Callable[[], None] | None = None
+        self._unsubscribe_member: Callable[[], None] | None = None
+        self._unsubscribe_activity: Callable[[], None] | None = None
+        self._unsubscribe_gv_signal: Callable[[], None] | None = None
+        self._unsubscribe_gv_req: Callable[[], None] | None = None
+        self._delay_unsubscribe: Callable[[], None] | None = None
+
+        self._value: int | None = None
+        self._last_rebind_time: float = 0.0
```

Comment:
Make it clear from the variable name it's a timstamp; end it with `_ts` or `_timestamp`, not `_time`